### PR TITLE
Revamp Change Pref Email & Add Success Pref Submission

### PR DESCRIPTION
**Pull request by Naoe-Adrian**
Email Revamp: Enhanced Faculty Preference Submission Flow  

#### Description  
We’ve updated the faculty preference submission email system to improve clarity and functionality. The following changes have been implemented:  
- A confirmation email is automatically sent to faculty upon successful submission of their preferences.  
- I dont know why the style change when i send it on actual email but i copy our font template and when i test it in my local its getting the right font
image
image

#### Action Required  
@Martinez-Emmanuel if you have sometime you can connect the success email send functionality to trigger immediately after faculty successfully submit their preferences.  

#### Testing  
- Ensure the confirmation email is sent only upon successful submission.  
- Verify email content matches the approved template.  

**Additional PR outputs December 30, 2024 API endpoint for retrieving faculty members with schedule reopening requests.**

1. Introduces a new API endpoint` (/api/request-notifications)` that fetches faculty members with pending schedule reopening requests. The response is in JSON format, containing an array of objects, each representing a faculty member with their ID, name, and a message indicating their request. This endpoint will be valuable for administrators to efficiently track and manage faculty schedule requests.

image

Feel free to share any feedback or report any issues.  
Thank you and happy holidays
---

### CODE REVIEW

**Comment by Naoe-Adrian** [comment]
@Malaluan-KylaRica  @Martinez-Emmanuel  kindly check this if this is correct thank you happy holidays

**Comment by Martinez-Emmanuel** [comment]
1. There is no need for a "successful submission" email anymore because our submission process has changed. As per PR [#69](https://github.com/PUPT-FLSS-ORG/PUPT-FLSS/pull/69), the preference submission is not done once, but over a period of time; as per PR [#84](https://github.com/PUPT-FLSS-ORG/PUPT-FLSS/pull/84), the "submit preferences" button that sends the bulk of preferences all at once is also gone. If you want to pursue this feature, change the message to "Preferences submission has now closed. Your preferences have been received and are currently under review." This email will then be sent ONCE **when the deadline for either all or individual faculty has passed.** You have to carefully modify our scheduled commands to make this work.

2. The reason why the custom fonts are not working is mainly due to the limitation of web fonts support for email clients (like Gmail). It's not your fault. I already tried to style our email to use the _PUP font_ for header and _Inter Tight_ for body for very obvious reasons and that's when I learned. There are workarounds for this but I didn't implement them because it's too trivial at this point of development.

3. The message for the `request-notifications` API must be "...reopen their preferences submission", not "schedule". I made a mistake and gave the wrong sample JSON in the group chat but I didn't think I needed to correct it since it was clear we were working with faculty preferences.

TL; DR - either discard "submission successful" email or modify it to be sent when the submission closes and correct the JSON response message for request notifications.

**Comment by Naoe-Adrian** [comment]
## 📬 Pull Request Revision: Add Single / Global Deadline Email Reminders and fixing Req notif 

### 📝 **Description**  
This commits **email reminders** for both **single faculty deadlines** and **global deadlines**, notifying faculty members **24 hours before their submission deadline**.

@Martinez-Emmanuel  I've put some 24 hrs. allowance because for me it's not making sense if we notify them that submission are close during that time I've think it's good to notify them 24hrs before the deadline for them to prepare thank you


### ✨ **What's New?**
1. **Single Deadline Reminder**  
   - Notifies individual faculty members whose `individual_deadline` is 24 hours away.  
   - Includes specific deadline date and a call-to-action button.  
   - Implemented via `NotifyFacultyDeadlineSingleJob`.

2. **Global Deadline Reminder**  
   - Notifies all faculty members whose `global_deadline` is 24 hours away.  
   - Provides a general reminder with submission details.  
   - Implemented via `NotifyGlobalFacultyDeadlineJob`.

3. **New Console Commands:**  
   - `preferences:check-deadline` – Checks and processes single deadlines.  
   - `preferences:check-global-deadline` – Checks and processes global deadlines.  

4. **API Endpoints:**  
   - `/api/notify-single-deadline` – Trigger single deadline notifications manually.  
   - `/api/notify-global-deadline` – Trigger global deadline notifications manually.  

5. **Email Templates:**  
   - `emails.faculty_deadline_warning`  
   - `emails.global_deadline_warning`

---

### 🛠️ **Implementation Details**
- Jobs are queued for better performance.  
- Commands are scheduled in the **Kernel.php** for automated checks.  
- Both APIs return detailed JSON responses for manual triggers.

---

### ✅ **How to Test**
1. **Run Console Commands:**  
   ```bash
   php artisan preferences:check-deadline
   ```
2. **Trigger APIs:**  

```
    // Test single faculty deadline notification
    Route::post('/test-faculty-notification', [EmailController::class, 'singleDeadlineNotification']);
    Route::get('/notify-global-deadline', [EmailController::class, 'notifyGlobalFacultyDeadline']);
```
   
3. **Verify Logs:**  
   ```bash
   tail -f storage/logs/laravel.log
   ```
4. **Check Email Inboxes:**  
   - Ensure single and global reminders are received with correct formatting.

---

### 📸 **Screenshots**

#### 🧑‍🎓 **Single Deadline Reminder Example**
image


#### 🌍 **Global Deadline Reminder Example**
image


image

---

### 🧪 **Testing Checklist**
- [x] Single deadline email sends correctly.  
- [x] Global deadline email sends correctly.  
- [x] Commands execute without errors.  
- [x] APIs return valid responses.  
- [x] Logs record successful notifications.

---

### 📦 **Dependencies**
- `Illuminate\Mail` for sending emails.  
- `Carbon\Carbon` for date handling.  
- Laravel Queues for job processing.

---

### 📚 **Documentation Updates**
- Added documentation for commands and API endpoints.  
- Updated README with testing instructions.

---

### 🚀 **Ready for Review**
Please review the code and test the features. Suggestions and feedback are welcome!

**Review by Martinez-Emmanuel — APPROVED** [review]
